case-lib/lib.sh

```diff
@@ -682,26 +682,10 @@ is_zephyr()
 
 is_firmware_file_zephyr()
 {
-    local firmware_name firmware_path znum platform
-    # get the FW name and path
-    # TODO: optimize this part after driver can expose the FW path to userspace via debugfs
-    if is_ipc4; then
-        firmware_name=dsp_basefw.bin
-        fw_mod_para=/sys/module/snd_sof_pci/parameters/fw_path
-        if [ -s "$fw_mod_para" ]; then
-            firmware_path=$(cat $fw_mod_para)
-        else
-            # # FIXME: the kernel driver should give us the FW path
-            # https://github.com/thesofproject/linux/issues/3867
-            die "Failed to get the IPC4 FW path."
-        fi
-    else # for IPC3
-        platform=$(sof-dump-status.py -p)
-        firmware_name=sof-$platform.ri
-        firmware_path=$(sof-dump-status.py -P)
-    fi
+    local firmware_path znum
 
-    znum=$(strings "/lib/firmware/$firmware_path/$firmware_name" | grep -c -i zephyr)
+    firmware_path=$(journalctl_cmd | grep "request_firmware" | awk -F ' ' '{print $(NF-1)}')
```

Comment:
Not a problem for CI but we're not sure what a developer may have done with their system, so maybe take the last one:

```suggestion
    firmware_path=$(journalctl_cmd -k | grep "request_firmware" | tail -n 1 | awk -F ' ' '{print $(NF-1)}')
```

Extra `-k` to be on safe side (hopefully journalctl accepts multiple `-k`)

Reply:
> Extra -k to be on safe side (hopefully journalctl accepts multiple -k)

`journalctl` does accept multiple `-k` so please add one here to be on the safe side. This function is in the "critical path" so we must make sure it never becomes slow no matter what `journalctl_cmd` becomes in the future.